src/__tests__/utils.spec.js

```diff
@@ -77,45 +77,69 @@ describe('utils', () => {
           });
         });
 
-        function testPrefixes(isRegex) {
-          isRegex = !!isRegex;
-          const simplePattern = isRegex?
-              '@duckduckgo\\.com' : '!duckduckgo.com';
+        function testPrefixes(pattern, expectedUrl, evilUrl) {
           return () => {
             it('should match url without path', () => {
               expect(
                   utils.matchesSavedMap(
-                      'https://duckduckgo.com',
+                      expectedUrl,
                       matchDomainOnly, {
-                        host: simplePattern,
+                        host: pattern,
                       })
               ).toBe(true);
             });
             it('should match url with path', () => {
               expect(
                   utils.matchesSavedMap(
-                      'https://duckduckgo.com/?q=search+me+baby',
+                    expectedUrl + '/?q=search+me+baby',
                       matchDomainOnly, {
-                        host: simplePattern,
+                        host: pattern,
                       })
               ).toBe(true);
             });
             let prefix = matchDomainOnly ? 'should not' : 'should';
-            let description = `${prefix} match url with pattern only in path`;
+            let description = `${pattern} ${prefix} match ${evilUrl}`;
             it(description, () => {
               expect(
                   utils.matchesSavedMap(
-                      'https://google.com/?q=duckduckgo',
+                      evilUrl,
                       matchDomainOnly, {
-                        host: simplePattern,
+                        host: pattern,
                       })
               ).toBe(!matchDomainOnly);
```

Comment:
This section's purpose is to make sure that `matchDomain=true` really only uses the domain for matching and not the URL path contain the domain. When `matchDomain=false` the url should match because the URL path contains the domain.

It looks like you repurposed the section to use `evilUrl` which makes sense when you put the domain in the URL's path. However, it becomes a lot less clear in the other cases.

I'd recommend explicitly adding an `it` for your test and reworking this part to add the domain of `expectedUrl` to the URL path.